django/db/models/sql/compiler.py

```diff
@@ -877,6 +877,11 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
                     if self._meta_ordering:
                         order_by = None
                 if having:
+                    if (
+                        self.connection.features.requires_group_by_with_having
+                        and not grouping
+                    ):
+                        result.append("GROUP BY TRUE")
```

Comment:
Given that `GROUP BY TRUE` is not universally going to do what we want I wonder if it would be better to have an operation (`connection.ops`) that returns the SQL that should be used when dealing with this case instead.

A bit like how do `ops.force_no_ordering()` above maybe we should do `ops.force_no_grouping()` and have it return an empty string except for SQLite.

I don't think it's strictly necessary here but it seems more aligned with how we provide other backend specific hooks.

Reply:
Hm yeah that sounds interesting…

So for sqlite:
```
    def force_group_by(self):
        return ["GROUP BY TRUE"]
```
with a default of:
```
    def force_group_by(self):
        return [""]
```